src/xdp-portal-impl.c

```diff
@@ -319,6 +314,33 @@ load_installed_portals (gboolean opt_verbose)
   implementations = g_list_sort (implementations, sort_impl_by_use_in_and_name);
 }
 
+void
+load_installed_portals (gboolean opt_verbose)
+{
+  const char * const *iter;
+  const char *portal_dir;
+
+  /* We need to override this in the tests */
+  portal_dir = g_getenv ("XDG_DESKTOP_PORTAL_DIR");
+  bool dir_set_from_env = (portal_dir == NULL);
+  if (portal_dir == NULL)
+    portal_dir = DATADIR "/xdg-desktop-portal/portals";
+
+  load_installed_portals_dir(opt_verbose, portal_dir);
+
+  if(dir_set_from_env)
+    return
+  
+  const char * const *dirs = g_get_system_data_dirs ();
```

Comment:
If we're going to load from `XDG_DATA_DIRS` then we should also try `XDG_DATA_HOME`, for completeness and to be consistent with the basedir spec.